File: src/ch04/code_3.py
from rich import print


def euclidean_algorithm(m: int, n: int) -> int:
    if n == 0:
        return m
    return euclidean_algorithm(m=n, n=m%n)


if __name__ == '__main__':
    result: int = euclidean_algorithm(m=51, n=15)
    print(result)

File: src/ch04/code_3.rs
fn euclidean_algorithm(m: i32, n: i32) -> i32 {
    if n == 0 {
        return m;
    }
    euclidean_algorithm(n, m%n)
}


fn main() {
    let result: i32 = euclidean_algorithm(51, 15);
    println!("{}", result);
}